modules/tinymce/src/themes/silver/main/ts/ui/core/color/Options.ts

```diff
@@ -84,18 +117,35 @@ const register = (editor: Editor): void => {
   });
 };
 
-const getColorCols = option('color_cols');
+const getColorCols = (editor: Editor, id: string): number => {
+  if (id === foregroundId) {
+    return option('color_cols_foreground')(editor);
+  } else if (id === backgroundId) {
+    return option('color_cols_background')(editor);
+  } else {
+    return option('color_cols')(editor);
```

Comment:
I think the `calcCols` need to move here getColorCols since it needs to compute different defaults depending on the format color map being used. So `getColorCols` could call getColors and do the calculation in the fallback.

Reply:
I'm not sure what you mean that it's needed to move here? It might be good to add a check that `color_map_background` or `color_map_foreground` are set before using those values, if we want them tied to those values, but the calculations should be just fine where they are?

Reply:
You are right. They fallback to the defaults for each option. So never mind. :+1: